lib/BackbeatConsumer.js

```diff
@@ -3,12 +3,13 @@ const kafka = require('node-rdkafka');
 const assert = require('assert');
 const async = require('async');
 const joi = require('joi');
+const Logger = require('werelogs').Logger;
+const zookeeper = require('node-zookeeper-client');
 
 const zookeeperHelper = require('./clients/zookeeper');
 const BackbeatProducer = require('./BackbeatProducer');
-const Logger = require('werelogs').Logger;
-const zookeeper = require('node-zookeeper-client');
 
+const replicationTopic = require('../conf/Config').extensions.replication.topic;
```

Comment:
Isn't this already available from the constructor as `this._topic`? This class has to be generic, enough to handle all topics.

Reply:
This conditional is needed I think, otherwise the canary will be sent for all topics. I'd have to check for canary messages on all consumers that extend BackbeatConsumer. Maybe I can change it so that the QueueProcessor triggers the canary being sent? And the QueueProcessor will check if its a replication group?

Reply:
May be not in this PR, this needs to be done cleanly.

Reply:
Instead I would rather have it configured via an option to the constructor. OK to do it later if time is too short.